projects/rocprofiler-systems/source/lib/core/trace_cache/metadata_registry.cpp

```diff
@@ -278,6 +278,89 @@ metadata_registry::get_kernel_symbol_list() const
     return result;
 }
 
+void
+metadata_registry::overwrite_cb_names(
+    std::initializer_list<
+        std::pair<rocprofiler_callback_tracing_kind_t, std::map<int, std::string_view>>>
+        categories_info)
+{
+    if(categories_info.size() == 0) return;
+
+    using category   = rocprofiler_callback_tracing_kind_t;
+    using operations = std::vector<std::string_view>;
+
+    auto category_names = std::vector<std::string_view>{};
+    auto modified_ops   = std::map<category, operations>{};
+
+    auto extract_operations = [&](category cat) -> operations {
+        auto        items           = m_callback_tracing_info.items();
+        const auto* target_category = items[static_cast<size_t>(cat)];
+
+        auto       operations_data = target_category->items();
+        operations operation_names;
+        operation_names.reserve(operations_data.size());
+
+        for(const auto& [op_idx, op_name] : operations_data)
+            operation_names.push_back(*op_name);
+
+        return operation_names;
+    };
+
+    // Store category names
+    category_names.resize(ROCPROFILER_CALLBACK_TRACING_LAST);
+    for(category i = ROCPROFILER_CALLBACK_TRACING_NONE;
+        i < ROCPROFILER_CALLBACK_TRACING_LAST;
+        i = static_cast<category>(static_cast<int>(i) + 1))
+    {
+        category_names[i] = m_callback_tracing_info.at(i);
+    }
+
+    // Process list
+    for(const auto& category_info : categories_info)
+    {
+        auto cat = category_info.first;
```

Comment:
```suggestion
        auto callback_kind = category_info.first;
```
Suggesting a clarifying variable name.